PurchasesCoreSwift/Networking/ETagManager.swift

```diff
@@ -0,0 +1,124 @@
+                                String(
+                                        format: Strings.network.could_not_find_cached_response_in_already_retried,
+                                        resultFromBackend
+                                )
+                        )
+                        return resultFromBackend
+                    } else {
+                        return nil
+                    }
+                }
+                return storedResponse
+            }
+            storeStatusCodeAndResponseIfNoError(
+                    for: request,
+                    statusCode: statusCode,
+                    responseObject: responseObject,
+                    eTag: eTagInResponse!)
+        }
+
+        return resultFromBackend
+    }
+
+    @objc public func clearCaches() {
+        self.userDefaults.removePersistentDomain(forName: ETagManager.suiteName)
+    }
+
+    private func shouldUseCachedVersion(responseCode: Int) -> Bool {
+        responseCode == HTTPStatusCodes.notModifiedResponseCode.rawValue
+    }
+
+    private func storedETagAndResponse(for request: URLRequest) -> ETagAndResponseWrapper? {
+        if let cacheKey = eTagDefaultCacheKey(for: request),
+            let value = userDefaults.object(forKey: cacheKey),
+            let data = value as? Data {
+                return ETagAndResponseWrapper(with: data)
+            }
+
+        return nil
+    }
+
+    private func getStoredHTTPResponse(for request: URLRequest) -> HTTPResponse? {
+        if let storedETagAndResponse = storedETagAndResponse(for: request) {
+            return HTTPResponse(
+                    statusCode: storedETagAndResponse.statusCode,
+                    responseObject: storedETagAndResponse.responseObject)
+        }
+
+        return nil
+    }
+
+    private func storeStatusCodeAndResponseIfNoError(for request: URLRequest,
+                                                     statusCode: Int,
+                                                     responseObject: [String: Any]?,
+                                                     eTag: String) {
+        if statusCode != HTTPStatusCodes.notModifiedResponseCode.rawValue &&
+                   statusCode < HTTPStatusCodes.internalServerError.rawValue &&
+                   responseObject != nil {
+            if let cacheKey = eTagDefaultCacheKey(for: request) {
+                let eTagAndResponse =
+                        ETagAndResponseWrapper(eTag: eTag, statusCode: statusCode, responseObject: responseObject!)
+                if let dataToStore = eTagAndResponse.asData() {
+                    userDefaults.set(dataToStore, forKey: cacheKey)
+                }
+            }
+        }
+    }
+
+    private func eTagDefaultCacheKey(for request: URLRequest) -> String? {
+        return request.url?.absoluteString
+    }
+
+    private static var suiteName: String {
+        let bundleID = Bundle.main.bundleIdentifier ?? ""
+        return bundleID + ".revenuecat.etags"
```

Comment:
should we handle the "no bundle identifier" case a bit more explicitly? 
I don't expect this to ever be empty outside of test suites, but perhaps we could have the `.` only be there if the bundleId isn't nil?